crates/iceberg-catalog/src/api/management/mod.rs

```diff
@@ -1492,30 +1553,125 @@ pub mod v1 {
     }
 
     /// Type of tabular
-    #[derive(Debug, Serialize, Clone, Copy, utoipa::ToSchema, strum::Display, PartialEq, Eq)]
+    #[derive(
+        Debug, Deserialize, Serialize, Clone, Copy, utoipa::ToSchema, strum::Display, PartialEq, Eq,
+    )]
     #[serde(rename_all = "kebab-case")]
     pub enum TabularType {
         Table,
         View,
     }
 
     #[derive(
-        Debug, Serialize, utoipa::ToSchema, Clone, Copy, PartialEq, Eq, strum_macros::Display,
+        Debug,
+        Deserialize,
+        Serialize,
+        utoipa::ToSchema,
+        Clone,
+        Copy,
+        PartialEq,
+        Eq,
+        strum_macros::Display,
     )]
     #[serde(rename_all = "kebab-case")]
     pub enum DeleteKind {
         Default,
         Purge,
     }
 
-    #[must_use]
-    pub fn api_doc<A: Authorizer>() -> utoipa::openapi::OpenApi {
+    pub fn api_doc<A: Authorizer>(
```

Comment:
If we build the APIDoc now dynamically, is there a place where we should cache it? Or should this method be cached?